src/__tests__/suggestions.js

```diff
@@ -196,9 +196,9 @@ test('escapes regular expressions in suggestion', () => {
   const {container} = renderIntoDocument(`
       <label for="superInput">inp-t lab^l w{th c+ars to esc\\pe</label>
       <input id="superInput" type="text" value="my super string +-('{}^$)" placeholder="should escape +-'(/" />
-      <p>
+      <span>
```

Comment:
Can you please explain why we needed this change? :)

Reply:
The `p` element here is used exclusively (or at least the choice of element) for a test asserting on the outcome of:

```js
getSuggestedQuery(container.querySelector('p')
```

to check that the suggestion properly escapes regular expressions.

ARIA 1.2 introduced a `paragraph` role which is the implicit role for the `p` element. With the bump of `aria-query` which is now tracking 1.2 (and some 1.3 features) this test started to fail as the suggestion became (the more natural) `getByRole("paragraph")`.

I’m not familiar with the history of why this particular assertion exists, but am guessing the handling of `.` and `()` in the suggestion generation was the focus over the specific element choice, thus opted for `span`.

TBH I was surprised the `span` worked as these now have an implicit role of `generic` which I’m fairly certain `aria-query` implements. Would need to dig into the code to see how the role detection works - if it is ignoring `generic` or using `dom-accessibility-api` (see https://github.com/eps1lon/dom-accessibility-api/pull/937) then it not suggesting `generic` makes sense.

In light of this, it’s probably safe to say this test may need changed again in future if/when `generic` roles are fully supported.

An alternative idea is to keep the `p` and instead opt to introduce another `p` element somewhere to force the suggestion to be more specific that just getting by role.

WDYT?

Reply:
Ah it seems there is `generic` specific logic, so think `span` could be long-lived here safely, see https://github.com/testing-library/dom-testing-library/blob/main/src/suggestions.js#L82